src/js/pouchPersisted.js

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.txt
+*/
+
+/* global fluid, floe, PouchDB */
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // Base grade for persistence of model components to Pouch
+    // Implementing grades should ensure that what's integral to
+    // reconstructing the particular instance is stored in the
+    // model
+    fluid.defaults("floe.dashboard.pouchPersisted", {
+        gradeNames: ["floe.dashboard.eventInTimeAware"],
+        model: {
+            // Stores the typename to assist in reconstruction
+            // of stored model components in another context
+            "persistenceInformation": {
+                "typeName": "{that}.typeName"
+            }
+        },
+        events: {
+            "onSetPouchId": null,
+            "onPouchDocStored": null,
+            "onPouchDocDeleted": null,
+            // Event signatures of this event should include the retrieved
+            // document
+            "onPouchDocRetrieved": null
+        },
+        listeners: {
+            "onCreate.setPouchId": {
+                funcName: "{that}.setPouchId",
+                args: "{that}"
+            }
+        },
+        dbOptions: {
+            // localName: "test",
+        },
+        // Implementing grades can use these invokers in combinations with
+        // events and listeners to store, retrieve and delete as
+        // appropriate
+        invokers: {
+            // Implementing grades or instances that want a different ID
+            // strategy should override this invoker
+            "setPouchId": {
+                funcName: "floe.dashboard.pouchPersisted.setPouchIdToCurrentTime",
+                args: "{that}"
+            },
+            "retrievePersisted": {
+                funcName: "floe.dashboard.pouchPersisted.retrievePersisted",
+                args: ["{that}", "{arguments}.0"]
+            },
+            "storePersisted": {
+                funcName: "floe.dashboard.pouchPersisted.storePersisted",
+                args: "{that}"
+            },
+            "deletePersisted": {
+                funcName: "floe.dashboard.pouchPersisted.deletePersisted",
+                args: "{that}"
+            }
+        }
+    });
+
+    // Set the ID to the timestamp to make filtering by time easier
+    floe.dashboard.pouchPersisted.setPouchIdToCurrentTime = function (that) {
+        that.model._id = that.model.timeEvents.created;
```

Comment:
This should never happen, and will be prohibited in a future revision of the framework. Always use the ChangeApplier to make model updates